CMakeLists.txt

```diff
@@ -1,10 +1,23 @@
 project(doppelkopf)
 cmake_minimum_required(VERSION 3.12)
 
+set(gtest_force_shared_crt ON)
 
-#add_subdirectory(libs/googletest)
 add_subdirectory(src)
-
 add_executable(doppelkopf ${SOURCE} ${CORE})
-#target_link_libraries(doppelkopf gtest_main)
-#add_test(NAME example_test COMMAND example)
\ No newline at end of file
+
+#tests are off by default, pass -DADD_TESTS=ON in to cmake to activate tests (and reset cache)
+option(ADD_TESTS "Set to ON to build with tests." OFF)
+
+IF(ADD_TESTS)
+
+    add_subdirectory(libs/googletest)
+    enable_testing()
+    include_directories(${gtest_SOURCE_DIR}/include ${COMMON_INCLUDES})
+
+    add_subdirectory(tests)
```

Comment:
When compiling, I had a naming conflict here, when building the target tests, it said `cannot open output file tests: is a directory`, which I think is because of this.

Reply:
will probably be solved by doing an out of source build with cmake